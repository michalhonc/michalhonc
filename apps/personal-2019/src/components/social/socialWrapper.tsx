import * as React from "react";

interface IProps {
  Icon: React.ElementType | string;
  profile: string;
  profileUrl: string;
  className: string;
}

export const SocialWrapper: React.FC<IProps> = ({
  Icon,
  profile,
  profileUrl,
  children,
  className,
}) => {
  return (
    <div className="Social-wrapper">
      <div className="Social-icon">
        <Icon />
      </div>
      <a href={profileUrl} className="underline Social-profile">
        {profile}
      </a>
      <div className={"Social-content " + className}>{children}</div>
    </div>
  );
};
