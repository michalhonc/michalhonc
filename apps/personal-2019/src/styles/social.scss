.Social {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;

  @include respond(thousand) {
    flex-direction: column;
    align-items: center;
  }

  &-wrapper {
    position: relative;
    border-radius: 2rem;
    border: 0.5rem solid var(--c-beta);
    padding: 3.5rem 3rem;
    width: 49rem;
    min-height: 70rem;
    height: 70rem;

    @include respond(phone) {
      max-width: 95%;
      height: initial;
      min-height: initial;
      padding: 3.5rem 1.5rem;
    }

    @include respond(thousand) {
      &:not(:first-of-type) {
        margin-top: 5rem;
      }
    }
  }

  &-icon {
    position: absolute;
    top:-4rem;
    left: 3rem;
    padding: 0.5rem;
    background-color: var(--c-white);

    & svg {
      width: 5rem;
      height: 5rem;
      fill: var(--c-alpha);
    }
  }

  &-profile {
    position: absolute;
    top: -3rem;
    left: 8rem + 1.5rem;
    font-size: 1.8rem;
    color: var(--c-beta);
    font-weight: bold;

    &:hover {
      text-decoration: underline;
    }
  }

  &-content {

  }
}
