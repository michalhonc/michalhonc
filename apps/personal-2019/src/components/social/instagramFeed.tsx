import * as React from "react";
import axios from "axios";

import { SocialWrapper } from "./socialWrapper";
import { State, useFetch } from "../../hooks/useFetch";
import Instagram from "../../images/instagram.svg";
import Like from "../../images/like.svg";
import Comment from "../../images/comment.svg";

interface IInstagram {
  big: string;
  small: string;
  url: string;
  caption: string;
  likes: number;
  comments: number;
  id: number;
}

export const InstagramFeed = () => {
  const { posts, state } = useFetch<IInstagram>("getInstagramPosts");

  if (state === State.Err) return null;

  return (
    <SocialWrapper
      Icon={Instagram}
      profile="@michalhonc.cz"
      profileUrl="https://www.instagram.com/michalhonc.cz/"
      className="InstagramFeed"
    >
      {state === State.Loading &&
        [...Array(6)].map((_, i) => (
          <a href="#" key={i}>
            <div className="InstagramFeed-imageWrapper--fake" />
          </a>
        ))}

      {posts.map((post: IInstagram) => (
        <a href={post.url} key={post.id}>
          <div className="InstagramFeed-imageWrapper">
            <img
              loading="lazy"
              width="200"
              height="200"
              src={post.small}
              alt="@michalhonc.cz Instagram profile"
            />
          </div>
          <div className="InstagramFeed-meta">
            {post.likes > 0 && (
              <span>
                {post.likes} <Like />
              </span>
            )}
            {post.comments > 0 && (
              <span>
                {post.comments} <Comment />
              </span>
            )}
          </div>
        </a>
      ))}
    </SocialWrapper>
  );
};
