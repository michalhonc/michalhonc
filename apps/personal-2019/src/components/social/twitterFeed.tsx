import * as React from "react";
import axios from "axios";

import { SocialWrapper } from "./socialWrapper";
import { State, useFetch } from "../../hooks/useFetch";
import Twitter from "../../images/twitter.svg";
import Like from "../../images/like.svg";
import Comment from "../../images/comment.svg";

interface ITwitter {
  createdAt: string;
  text: string;
  small: string | null;
  thumb: string | null;
  id: number;
  url: string;
}

export const TwitterFeed = () => {
  const { posts, state } = useFetch<ITwitter>("getTwitterPosts");

  if (state === State.Err) return null;

  return (
    <SocialWrapper
      Icon={Twitter}
      profile="@michalhonc"
      profileUrl="https://twitter.com/Michalhonc"
      className="TwitterFeed"
    >
      {state === State.Loading && <span>loading</span>}
      {posts.map((post: ITwitter) => (
        <a href={post.url} key={post.id}>
          <div
            className="TwitterFeed-tweet"
            dangerouslySetInnerHTML={{ __html: underlineHashtags(post.text) }}
          />
          <div className="TwitterFeed-imageWrapper">
            {post.small && (
              <img
                loading="lazy"
                width="120"
                height="80"
                src={post.small}
                alt="@michalhonc Twitter profile"
              />
            )}
          </div>
        </a>
      ))}
    </SocialWrapper>
  );
};

function underlineHashtags(text: string) {
  return text
    .split(" ")
    .map((str) =>
      str.startsWith("#") || str.startsWith("@")
        ? `<span class="underline">${str}</span>`
        : str
    )
    .join(" ");
}
