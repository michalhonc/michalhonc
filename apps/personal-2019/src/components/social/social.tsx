import * as React from "react";

import { InstagramFeed } from "./instagramFeed";
import { TwitterFeed } from "./twitterFeed";

export const Social = () => {
  return (
    <section id="portoflio" className="Container Social">
      <InstagramFeed />
      <TwitterFeed />
    </section>
  );
};
