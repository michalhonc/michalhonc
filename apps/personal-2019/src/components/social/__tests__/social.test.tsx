import * as React from "react";
import renderer from "react-test-renderer";
import { Social } from "../social";

describe("social root", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Social />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
