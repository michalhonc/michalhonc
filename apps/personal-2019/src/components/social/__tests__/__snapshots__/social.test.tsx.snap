// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`social root renders correctly 1`] = `
<section
  className="Container Social"
  id="portoflio"
>
  <div
    className="Social-wrapper"
  >
    <div
      className="Social-icon"
    >
      <test-file-stub />
    </div>
    <a
      className="underline Social-profile"
      href="https://www.instagram.com/michalhonc.cz/"
    >
      @michalhonc.cz
    </a>
    <div
      className="Social-content InstagramFeed"
    />
  </div>
  <div
    className="Social-wrapper"
  >
    <div
      className="Social-icon"
    >
      <test-file-stub />
    </div>
    <a
      className="underline Social-profile"
      href="https://twitter.com/Michalhonc"
    >
      @michalhonc
    </a>
    <div
      className="Social-content TwitterFeed"
    />
  </div>
</section>
`;
